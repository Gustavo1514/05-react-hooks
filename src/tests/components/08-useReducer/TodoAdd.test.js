const { shallow } = require('enzyme')
const {
  default: TodoAdd
} = require('../../../components/08-useReducer/TodoAdd')

describe('pruebas en <TodoAdd/>', () => {
  const handleAddTodo = jest.fn()
  const wrapper = shallow(<TodoAdd handleAddTodo={handleAddTodo} />)
  test('debe mostrarse correctamente ', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('No debe llmar handleAddTodo ', () => {
    // wrapper.find('form').simulate('submit', { preventDefault () {} })

    const formSubmit = wrapper.find('form').prop('onSubmit')
    formSubmit({ preventDefault () {} })
    expect(handleAddTodo).toHaveBeenCalledTimes(0)
  })

  test('debe llamar la funcion handleAddTodo', () => {
    const value = 'adios Mundo'
    let input = wrapper.find('input')
    input.simulate('change', {
      target: {
        name: 'description',
        value
      }
    })
    const formSubmit = wrapper.find('form').prop('onSubmit')
    formSubmit({ preventDefault () {} })
    expect(handleAddTodo).toHaveBeenCalledTimes(1)
    expect(handleAddTodo).toHaveBeenCalledWith({
      id: expect.any(Number),
      description: value,
      done: false
    })
    expect(wrapper.find('input').prop('value')).toBe('')
  })
})
