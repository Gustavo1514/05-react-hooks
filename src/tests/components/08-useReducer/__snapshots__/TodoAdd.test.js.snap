// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <TodoAdd/> debe mostrarse correctamente  1`] = `
<Fragment>
  <h4>
    Agregar TODO
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      id=""
      name="description"
      onChange={[Function]}
      placeholder="aprender ..."
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary mt-2"
      type="submit"
    >
      Agregar
    </button>
  </form>
</Fragment>
`;
